tools/pylib/boutdata/data.py

```diff
@@ -554,10 +557,15 @@ def __init__(self, path=".", prefix="BOUT.dmp", suffix=None, caching=False,
         # Available variables
         self.varNames = []
         self.dimensions = {}
-        self.evolvingVariableNames = []
+        self.evolvingVariables = []
+        self._shiftAngle = None # used for getShifted()/getUnshifted() methods, calculated from self['zShift'] when first needed
 
         with DataFile(latest_file) as f:
-            npes = f.read("NXPE")*f.read("NYPE")
+            self.attributes = f.attributes()
+            try:
+                npes = f.read("NXPE")*f.read("NYPE")
+            except KeyError:
+                npes = self._attributes["NXPE"]*self._attributes["NYPE"]
```

Comment:
This is `_attributes`, but `attributes` above. Are they supposed to be different variables?
Also, are `NXPE/NYPE` actually file attributes rather than variables?